test/merger/test_single_qubit_gates_merger.py

```diff
@@ -0,0 +1,268 @@
+import math
+
+import pytest
+
+from opensquirrel import Circuit, CircuitBuilder
+from opensquirrel.default_instructions import Ry, Rz
+from opensquirrel.ir import Bit, BlochSphereRotation, Float, Qubit
+from opensquirrel.passes.merger import SingleQubitGatesMerger
+from opensquirrel.passes.merger.general_merger import rearrange_barriers
+from test.ir_equality_test_base import modify_circuit_and_check
+
+
+@pytest.fixture(name="merger")
+def merger_fixture() -> SingleQubitGatesMerger:
+    return SingleQubitGatesMerger()
+
+
+def test_single_gate(merger: SingleQubitGatesMerger) -> None:
+    builder1 = CircuitBuilder(1)
+    builder1.Ry(0, Float(1.2345))
+    circuit = builder1.to_circuit()
+
+    builder2 = CircuitBuilder(1)
+    builder2.Ry(0, Float(1.2345))
+    expected_circuit = builder2.to_circuit()
+
+    modify_circuit_and_check(circuit, merger.merge, expected_circuit)
+
+    # Check that when no fusion happens, generator and arguments of gates are preserved.
+    assert isinstance(circuit.ir.statements[0], BlochSphereRotation)
+    assert circuit.ir.statements[0].generator == Ry
+    assert circuit.ir.statements[0].arguments == (Qubit(0), Float(1.2345))
```

Comment:
Do we still need these asserts? Isn't `modify_circuit_and_check` enough?

Reply:
Yes, `modify_circuit_and_check` does not check that what comes after fusion is a `BlochSphereRotation`, that a single single-qubit gate is not altered, and that its arguments are still the same. 

I mean, these asserts could be condensed to:
```suggestion
    assert circuit.ir.statements == [Ry(0, 1.2345)]
```

Reply:
That last line is done in `modify_circuit_and_check`, in `assert circuit == expected_circuit`. That equality checks that the IR's are equal.